### PR TITLE
Add spike in counts for Zeisel data

### PR DESCRIPTION
**Pull request by Alanocallaghan**
As ~~threatened~~ promised on Bioc slack.

This uses a table each from Svensson's power analysis paper and the spike in mix concs from Thermo. https://www.nature.com/articles/nmeth.4220

Should I use BiocFileCache for the tables? I think I trust Nature to be stable, but I'd be inclined to include the file from Thermo in the repo, since manufacturers can be pretty bad for randomly restructuring their websites.

### CODE REVIEW

**Comment by LTLA** [comment]
> Should I use BiocFileCache for the tables?

Yes. 

The only modifications necessary should occur in `make-zeisel-brain-data.Rmd` and `make-zeisel-brain-metadata.R`. There shouldn't be added data files nor `scRNAseq:::.calculate_spikeins`.

**Comment by Alanocallaghan** [comment]
Cool thanks, figured it wasn't the right approach but I'm not familiar with the bioc hub tools. I'll change this evening. once you're happy with the setup for zeisel I'll see about adding others

**Comment by LTLA** [comment]
Several points, in order:

- Why is there `R/ERCCSpikeInConcentrations.R` and `R/SpikeInConcentrations.R`?

- I just realized that we don't even need a `make-spike-in-concentrations-data.Rmd`. Just take the file as is and put it on ExperimentHub. (We will need to check if there are any licence issues, but probably no one will care.) The file is small enough that the parsing can be done in `ERCCSpikeInConcentrations()` without the need for more fiddling.

- In a similar vein, put the inclusion of spike-in information in `ZeiselBrainData()` rather than `make-zeisel-brain-data.Rmd`, which saves me the trouble of having to upload a new version.

Don't bother force pushing, I'm going to squash all the commits anyway.

**Comment by Alanocallaghan** [comment]
It's a mess because I didn't get to finishing edits (ran out of hours.on Sunday evening). Sounds good I'll make those changes today

**Comment by Alanocallaghan** [comment]
>Just take the file as is and put it on ExperimentHub.

Does this entail making an ExperimentHub package, or do you mean using the existing link and adding metadata to EH? If the latter surely just BiocFileCache'ing is functionally the same.

**Comment by LTLA** [comment]
We will stick it onto EHub as its own entry so that it persists even after ThermoFisher changes its link.

**Comment by Alanocallaghan** [comment]
Fair dues. Should I ping Lori about it then, or how does one upload?

**Comment by LTLA** [comment]
I'll handle it, I've got a whole bunch of EHub-related stuff to do anyway.

**Comment by LTLA** [comment]
But do clean up the remainders of what I suggested.

**Comment by Alanocallaghan** [comment]
Yeah will do of course. Thanks for the feedback

On 4 Mar 2020 5:50 am, Aaron Lun <notifications@github.com> wrote:

But do clean up the remainders of what I suggested.

**Comment by Alanocallaghan** [comment]
Fixed, though may need to fix the EHub download code if the metadata doesn't match.

File: DESCRIPTION [hunk 1]
```diff
@@ -1,7 +1,7 @@
 Package: scRNAseq
 Title: Collection of Public Single-Cell RNA-Seq Datasets
-Version: 2.1.6
-Date: 2020-02-14
+Version: 2.2.0
```
Comment (LTLA) [inline DESCRIPTION]: Don't do that, this is bumped during BioC release.

File: R/ERCCSpikeInConcentrations.R [hunk 2]
```diff
@@ -14,21 +17,24 @@
 #' 
 #' @export
 #' @importFrom SingleCellExperiment splitAltExps
-ERCCSpikeInConcentrations <- function(volume_nl, dilution, genes, mix1 = TRUE) {
+ERCCSpikeInConcentrations <- function(genes = NULL, volume_nl = NULL, dilution = NULL) {
```
Comment (LTLA) [inline R/ERCCSpikeInConcentrations.R]: No need for `volume_nl`, just have `volume` and use the documentation to mention what the unit is.

File: R/ERCCSpikeInConcentrations.R [hunk 3]
```diff
@@ -5,6 +5,9 @@
 #' All data are downloaded from ExperimentHub and cached for local re-use.
 #' Specific resources can be retrieved by searching for \code{scRNAseq/ercc-spike-in-concentrations}.
 #'
+#' @param genes A character vector specifying the ERCC spike-in genes used.
+#' @param volume_nl Volume (in nanoliters) of ERCC spike-in mixture.
```
Comment (LTLA) [inline R/ERCCSpikeInConcentrations.R]: "Numeric scalar specifying the volume..."

File: R/ERCCSpikeInConcentrations.R [hunk 4]
```diff
@@ -5,6 +5,9 @@
...
+#' @param dilution The ratio of ERCC mixture to total well content.
```
Comment (LTLA) [inline R/ERCCSpikeInConcentrations.R]: "Numeric scalar containing the ratio..."

File: DESCRIPTION (hunk 1 above)
Comment (Alanocallaghan) [reply DESCRIPTION]: Date as well?

Comment (LTLA) [reply DESCRIPTION]: The new date is fine.

File: R/ZeiselBrainData.R [hunk 5]
```diff
@@ -44,6 +44,12 @@ ZeiselBrainData <- function(ensembl=FALSE, location=TRUE) {
     status[status=="mito"] <- "endogenous"
     sce <- splitAltExps(sce, status, "endogenous")
 
+    spikedata <- ERCCSpikeInConcentrations(
+        genes = rownames(altExp(sce, "ERCC")),
+        volume_nl = 9,
```
Comment (LTLA) [inline R/ZeiselBrainData.R]: This'll need to be updated to `volume`.

**Comment by Alanocallaghan** [comment]
Should be all good now, I've checked locally and both user-facing changes seem to work

**Pull request merged**